### ISSUE TITLE
Invalid transformation format:AES/256/CBC/PKCS5Padding : Interoperabilty between Python and Java SDKs

### ISSUE THREAD

**Issue by darpk** [aws/aws-dynamodb-encryption-python#145]
I tried a small custom application which encrypts and writes records to DDB using the Python SDK. The code (copied from [here](https://docs.aws.amazon.com/dynamodb-encryption-client/latest/devguide/most-recent-provider.html)) is as following:

```
    aws_kms_cmp = AwsKmsCryptographicMaterialsProvider(key_id=aws_cmk_id)

    meta_table_name = 'metaStoreTable'

    # A name for the Most Recent Provider; used to
    # identify its CMPs in provider store storage
    material_name = 'testMRP'

    # Create an internal DynamoDB table resource for the MetaStore
    meta_table = boto3.resource('dynamodb').Table(meta_table_name)

    # Create the MetaStore with the Direct KMS Provider
    meta_store = MetaStore(
        table=meta_table,
        materials_provider=aws_kms_cmp
    )

    # Create a Most Recent Provider using the MetaStore
    most_recent_cmp = MostRecentProvider(
        provider_store=meta_store,
        material_name=material_name,
        version_ttl=60.0  # Check for a new version every 60 seconds
    )

    actions = AttributeActions(
        default_action=CryptoAction.SIGN_ONLY,
        attribute_actions={'cSnippet': CryptoAction.ENCRYPT_AND_SIGN}
    )

    encrypted_table = EncryptedTable(
        table=table,
        materials_provider=most_recent_cmp,
        attribute_actions=actions
    )
    encrypted_table.put_item(Item=result)
```

Java code to read records is below (taken from [here](https://docs.aws.amazon.com/dynamodb-encryption-client/latest/devguide/most-recent-provider.html))

```
            final AmazonDynamoDB dynamoDbClient = AmazonDynamoDBClientBuilder.standard().withRegion(REGION).build();

            final AWSKMS awskms = getOrCreateKMSClientBuilder().build();
            final String kmsKey = System.getenv(DYNAMODB_KMS_KEY); //same key arn as Python code

            final DirectKmsMaterialProvider directKmsMaterialProvider = new DirectKmsMaterialProvider(awskms, kmsKey);
            final DynamoDBEncryptor dynamoDBEncryptor = DynamoDBEncryptor.getInstance(directKmsMaterialProvider);

            final String keyTableName = "metaStoreTable"; //for MetaStore's internal table
            final long ttlInMillis = 60000;
            final String materialName = "testMRP";
            final MetaStore metaStore = new MetaStore(dynamoDbClient, keyTableName, dynamoDBEncryptor);
            final MostRecentProvider cmp = new MostRecentProvider(metaStore, materialName, ttlInMillis);

            final DynamoDBMapperConfig mapperConfig = DynamoDBMapperConfig.builder()
                    .withSaveBehavior(DynamoDBMapperConfig.SaveBehavior.PUT)
                    .withConsistentReads(DynamoDBMapperConfig.ConsistentReads.EVENTUAL)
                    .build();

            final DynamoDBMapper dynamoDBMapper = new DynamoDBMapper(dynamoDbClient,
                    mapperConfig, new AttributeEncryptor(cmp));

            dynamoDBMapper.queryPage(tableName, queryExpression);
```

Exception Traceback:

```
Caused by: java.lang.IllegalArgumentException: Algorithm does not exist
--
at com.amazonaws.services.dynamodbv2.datamodeling.encryption.DynamoDBEncryptor.lambda$static$0(DynamoDBEncryptor.java:69) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660) ~[?:1.8.0_201]
at com.amazonaws.services.dynamodbv2.datamodeling.encryption.DynamoDBEncryptor.getBlockSize(DynamoDBEncryptor.java:396) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
at com.amazonaws.services.dynamodbv2.datamodeling.encryption.DynamoDBEncryptor.actualDecryption(DynamoDBEncryptor.java:380) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
at com.amazonaws.services.dynamodbv2.datamodeling.encryption.DynamoDBEncryptor.decryptRecord(DynamoDBEncryptor.java:283) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
at com.amazonaws.services.dynamodbv2.datamodeling.AttributeEncryptor.untransform(AttributeEncryptor.java:100) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
... 26 more
Caused by: java.security.NoSuchAlgorithmException: Invalid transformation format:AES/256/CBC/PKCS5Padding
at javax.crypto.Cipher.tokenizeTransformation(Cipher.java:321) ~[?:1.8.0_201]
at javax.crypto.Cipher.getTransforms(Cipher.java:428) ~[?:1.8.0_201]
at javax.crypto.Cipher.getInstance(Cipher.java:502) ~[?:1.8.0_201]
at com.amazonaws.services.dynamodbv2.datamodeling.encryption.DynamoDBEncryptor.lambda$static$0(DynamoDBEncryptor.java:66) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660) ~[?:1.8.0_201]
at com.amazonaws.services.dynamodbv2.datamodeling.encryption.DynamoDBEncryptor.getBlockSize(DynamoDBEncryptor.java:396) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
at com.amazonaws.services.dynamodbv2.datamodeling.encryption.DynamoDBEncryptor.actualDecryption(DynamoDBEncryptor.java:380) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
at com.amazonaws.services.dynamodbv2.datamodeling.encryption.DynamoDBEncryptor.decryptRecord(DynamoDBEncryptor.java:283) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]
at com.amazonaws.services.dynamodbv2.datamodeling.AttributeEncryptor.untransform(AttributeEncryptor.java:100) ~[aws-dynamodb-encryption-java-1.14.1.jar:?]

```

**Python SDK version for the client: 1.2.0**
**Java SDK version for the client: 1.14.1**

**Comment by robin-aws** [comment]
Hi there! Thanks for reaching out.

This appears to be an incompatibility between the Java and Python DynamoDB Encryption Clients. The actual payloads are compatible, but the two libraries use different strings in the metadata to represent the same choice of algorithm (`AES` in Java vs. `AES/256`), and the Java library doesn't understand the string Python produces.

We should be able to address this with a patch on the Java side (https://github.com/aws/aws-dynamodb-encryption-java/).

**Comment by lavaleri** [comment]
Hi,

This is fixed as of the 1.15.1 and 2.0.1 releases of the DynamoDB Encryption Client for Java.

Thank you for the report!
